pkg/kube/pod_test.go

```diff
@@ -118,3 +118,349 @@ func (s *PodSuite) TestGetPodLogs(c *C) {
+					},
+				},
+				RestartPolicy: v1.RestartPolicyOnFailure,
+				Volumes: []v1.Volume{
+					{
+						Name: "data",
+						VolumeSource: v1.VolumeSource{
+							PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+								ClaimName: "default-pvc",
+							},
+						},
+					},
+				},
+			},
+		},
+
+		// Add NodeSelector and Tolerations. Override container command
+		{
+			BlueprintPodSpecs: map[string]interface{}{
+				"nodeSelector": map[string]interface{}{
+					"selector-key": "selector-value",
+				},
+				"tolerations": []map[string]interface{}{
+					{
+						"key":      "taint-key",
+						"operator": "Equal",
+						"value":    "taint-value",
+						"effect":   "NoSchedule",
+					},
+				},
+			},
+			ActionsetPodSpecs: map[string]interface{}{
+				"containers": []map[string]interface{}{
+					{
+						"name":    "container",
+						"command": []string{"echo", "override command"},
+					},
+				},
+			},
+			Expected: v1.PodSpec{
+				NodeSelector: map[string]string{
+					"selector-key": "selector-value",
+				},
+				Tolerations: []v1.Toleration{
+					{
+						Key:      "taint-key",
+						Operator: v1.TolerationOpEqual,
+						Value:    "taint-value",
+						Effect:   v1.TaintEffectNoSchedule,
+					},
+				},
+				Containers: []v1.Container{
+					{
+						Name:            "container",
+						Image:           "kanisterio/kanister-tools:0.21.0",
+						Command:         []string{"echo", "override command"},
+						ImagePullPolicy: v1.PullPolicy(v1.PullAlways),
+						VolumeMounts: []v1.VolumeMount{
+							{
+								Name:      "data",
+								MountPath: "/var/lib/data",
+							},
+						},
+					},
+				},
+				RestartPolicy: v1.RestartPolicyOnFailure,
+				Volumes: []v1.Volume{
+					{
+						Name: "data",
+						VolumeSource: v1.VolumeSource{
+							PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+								ClaimName: "default-pvc",
+							},
+						},
+					},
+				},
+			},
+		},
+
+		// Override container command
+		{
+			BlueprintPodSpecs: map[string]interface{}{
+				"containers": []map[string]interface{}{
+					{
+						"name":    "container",
+						"command": []string{"echo", "override command"},
+					},
+				},
+			},
+			ActionsetPodSpecs: nil,
+			Expected: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:            "container",
+						Image:           "kanisterio/kanister-tools:0.21.0",
+						Command:         []string{"echo", "override command"},
+						ImagePullPolicy: v1.PullPolicy(v1.PullAlways),
+						VolumeMounts: []v1.VolumeMount{
+							{
+								Name:      "data",
+								MountPath: "/var/lib/data",
+							},
+						},
+					},
+				},
+				RestartPolicy: v1.RestartPolicyOnFailure,
+				Volumes: []v1.Volume{
+					{
+						Name: "data",
+						VolumeSource: v1.VolumeSource{
+							PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+								ClaimName: "default-pvc",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Run tests
```

Comment:
@PrasadG193 unit tests look great, thanks for taking care of this. just a small request: can we also show one scenario that demonstrates the priority of Actionset spec or blueprint spec , when both have same overrider field with different value ?